components/telemetry-operator/controllers/logpipeline_controller.go

```diff
@@ -0,0 +1,398 @@
+		log.Info("Updated fluent bit configuration")
+		if err := r.Update(ctx, &logPipeline); err != nil {
+			log.Error(err, "Cannot update LogPipeline")
+			return ctrl.Result{}, err
+		}
+
+		if err := r.deleteFluentBitPods(ctx, log); err != nil {
+			log.Error(err, "Cannot delete fluent bit pods")
+			return ctrl.Result{}, err
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// Get ConfigMap from Kubernetes API or create new one if not existing.
+func (r *LogPipelineReconciler) getOrCreateConfigMap(ctx context.Context, name types.NamespacedName) (corev1.ConfigMap, error) {
+	var cm corev1.ConfigMap
+	if err := r.Get(ctx, name, &cm); err != nil {
+		if errors.IsNotFound(err) {
+			cm = corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name.Name,
+					Namespace: name.Namespace,
+				},
+			}
+			if err := r.Create(ctx, &cm); err != nil {
+				return cm, err
+			}
+		} else {
+			return cm, err
+		}
+	}
+	return cm, nil
+}
+
+// Synchronize LogPipeline with ConfigMap of FluentBit sections (Input, Filter and Output).
+func (r *LogPipelineReconciler) syncSectionsConfigMap(ctx context.Context, logPipeline *telemetryv1alpha1.LogPipeline) (bool, error) {
+	log := log.FromContext(ctx)
+	cm, err := r.getOrCreateConfigMap(ctx, r.FluentBitSectionsConfigMap)
+	if err != nil {
+		return false, err
+	}
+	cmKey := logPipeline.Name + ".conf"
+
+	// Add or remove Fluent Bit configuration sections
+	if logPipeline.DeletionTimestamp != nil {
+		if cm.Data != nil && controllerutil.ContainsFinalizer(logPipeline, sectionsConfigMapFinalizer) {
+			log.Info("Deleting fluent bit config")
+			delete(cm.Data, cmKey)
+			controllerutil.RemoveFinalizer(logPipeline, sectionsConfigMapFinalizer)
+		}
+	} else {
+		fluentBitConfig := mergeFluentBitConfig(logPipeline)
+		if cm.Data == nil {
+			data := make(map[string]string)
+			data[cmKey] = fluentBitConfig
+			cm.Data = data
+		} else {
+			if oldConfig, hasKey := cm.Data[cmKey]; hasKey && oldConfig == fluentBitConfig {
+				// Nothing changed
+				return false, nil
+			}
+			cm.Data[cmKey] = fluentBitConfig
+		}
+		controllerutil.AddFinalizer(logPipeline, sectionsConfigMapFinalizer)
+	}
+
+	// Update ConfigMap
+	if err := r.Update(ctx, &cm); err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+// Synchronize LogPipeline with ConfigMap of FluentBit parsers (Parser and MultiLineParser).
+func (r *LogPipelineReconciler) syncParsersConfigMap(ctx context.Context, logPipeline *telemetryv1alpha1.LogPipeline) (bool, error) {
+	log := log.FromContext(ctx)
+	cm, err := r.getOrCreateConfigMap(ctx, r.FluentBitParsersConfigMap)
+	if err != nil {
+		return false, err
+	}
+	cmKey := "parsers.conf"
+
+	// Add or remove Fluent Bit configuration parsers
+	if logPipeline.DeletionTimestamp != nil {
+		if cm.Data != nil && controllerutil.ContainsFinalizer(logPipeline, parserConfigMapFinalizer) {
+			log.Info("Deleting fluent bit parsers config")
+			delete(cm.Data, cmKey)
```

Comment:
This may also delete parsers from other LogPipeline resources. We have to regenerate the configuration from all remaining LogPipelines in this case.